### PR TITLE
respond to config change for terminal suggestions

### LINKED ISSUE #235071: Enabling terminal completion requires restart
Testing #235021

* enable  `"terminal.integrated.suggest.enabled": true,  "terminal.integrated.suggest.enableExtensionCompletions": true,`
* :bug: no completions in terminal
* reload the window, completions appear

### PR DESCRIPTION
**Pull request by meganrogge**
fix #235071

### CODE REVIEW

File: src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts [hunk 1]
```diff
@@ -61,6 +61,25 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 			this._pwshAddon?.dispose();
 		}));
 		this._terminalSuggestWidgetVisibleContextKey = TerminalContextKeys.suggestWidgetVisible.bindTo(this._contextKeyService);
+		this.add(this._configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(TerminalSuggestSettingId.EnableExtensionCompletions) || e.affectsConfiguration(TerminalSuggestSettingId.Enabled)) {
+				const extensionCompletionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enableExtensionCompletions;
+				const completionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enabled;
+				if (!extensionCompletionsEnabled) {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts:68]: Could we make this `!extensionCompletionsEnabled || !completionEnabled` instead and remove it from the lower if?

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts]: makes it a little more confusing imo...

Comment (meganrogge) [reply src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts]: oh, I see what you mean, fixed

File: src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts [hunk 2]
```diff
@@ -61,6 +61,22 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
...
 		}));
 		this._terminalSuggestWidgetVisibleContextKey = TerminalContextKeys.suggestWidgetVisible.bindTo(this._contextKeyService);
+		this.add(this._configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(TerminalSuggestSettingId.EnableExtensionCompletions) || e.affectsConfiguration(TerminalSuggestSettingId.Enabled)) {
+				const extensionCompletionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enableExtensionCompletions;
+				const completionsEnabled = this._configurationService.getValue<ITerminalSuggestConfiguration>(terminalSuggestConfigSection).enabled;
+				if (!extensionCompletionsEnabled || !completionsEnabled) {
+					this._addon.clear();
+					this._pwshAddon.clear();
+				}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts:71]: Isn't the right condition now this?

```ts
				if (!extensionCompletionsEnabled || !completionsEnabled) {
					this._addon.clear();
				}
				if (!completionsEnabled) {
					this._pwshAddon.clear();
				}
```

Comment (meganrogge) [reply src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts:71]: we can clear both bc below will create if neeeded

Comment (meganrogge) [reply src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts:71]: oh nvm, I thought we checked those settings in `_loadAddons`, but we don't.

**Review by Tyriar — APPROVED** [review]

**Pull request merged**